### ISSUE TITLE
OAuth2 Resource Server docs not in sync - authorityPrefix can't be set to ""

### ISSUE THREAD

**Issue by judomu** [spring-projects/spring-security#8421]
In spring-security documentation (https://docs.spring.io/spring-security/site/docs/current/reference/html5/):

> Or, you can remove the prefix altogether by calling JwtGrantedAuthoritiesConverter#setAuthorityPrefix("").

But that will not work

```
	public void setAuthorityPrefix(String authorityPrefix) {
		Assert.hasText(authorityPrefix, "authorityPrefix cannot be empty");
		this.authorityPrefix = authorityPrefix;
	}
```

So the documentation is wrong or the implementation, imho the option to set authorityPrefix to "" makes sense to me.

**Comment by rwinch** [comment]
Thanks for the report. I think it is the implementation that is broken. Would you like to submit a PR to fix it?

**Comment by judomu** [comment]
Sure, I will provide the fix

**Comment by rwinch** [comment]
Thanks @judomu! The issue is yours

**Comment by rwinch** [comment]
This is now merged into master and 5.2.x
